bots/image-download

```diff
@@ -205,6 +206,21 @@ def committed_target(image):
 
     return dest
 
+def wait_lock(target):
+    lockfile = os.path.join(tempfile.gettempdir(), ".cockpit-test-resources", os.path.basename(target) + ".lock")
+    # we need to keep the lock fd open throughout the entire runtime, so remember it in a global-scoped variable
+    wait_lock.f = open(lockfile, "w")
+    for retry in range(100):
+        try:
+            fcntl.flock(wait_lock.f, fcntl.LOCK_NB | fcntl.LOCK_EX)
+            return
+        except BlockingIOError as e:
+            print("Waiting for concurrent image-download of %s..." % os.path.basename(target))
+            time.sleep(10)
+    else:
+        sys.stderr.write("image-download: timed out waiting for concurrent downloads of %s\n" % target)
+        sys.exit(1)
```

Comment:
Do you mind changing this to raise a `TimeoutError`? The only invocation of this is wrapped in a `try` block already and I dislike `sys.exit()` calls in random locations.

Reply:
Indeed, well spotted. This looks still fine:
```
Waiting for concurrent image-download of cirros-0.3.5-i386-disk.img...
image-download: image-download: timed out waiting for concurrent downloads of /srv/vm/cockpit-test-data/images/cirros-0.3.5-i386-disk.img
```
(Tested by reducing the retries to 2).

I took the opportunity to fix the typo in the commit message and adding a closes. I also bumped to 360 retries to have a nice round one-hour total timeout.